packages/jazz-tools/src/tools/tests/ContextManager.test.ts

```diff
@@ -323,11 +326,11 @@ describe("ContextManager", () => {
         root: AccountRoot,
         profile: co.profile(),
       })
-      .withMigration(async (account) => {
-        account.root = AccountRoot.create({
+      .withMigration(async (account: Account) => {
+        account.$jazz.set("root", {
           value: "Hello",
         });
-        lastRootId = account.root.$jazz.id;
+        lastRootId = account.root!.$jazz.id;
```

Comment:
The caveat from using `$jazz.set` instead of direct assignments is that TS can't know that the property has been set. I tried to get this working by adding an [assert condition](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions) to `Account.$jazz.set`, but I don't think it's possible to narrow the type of `account` when calling a method in `account.$jazz`